app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -421,14 +422,52 @@ public void onFrameAvailable(SurfaceTexture surfaceTexture) {
         });
     }
 
+    @Keep
+    @SuppressWarnings("unused")
+    void dispatchCreateWidgetLayer(final int aHandle, final Surface aSurface, final int aWidth, final int aHeight) {
+        runOnUiThread(() -> {
+            final Widget widget = mWidgets.get(aHandle);
+            if (widget == null) {
+                Log.e(LOGTAG, "Widget " + aHandle + " not found");
+                return;
+            }
+
+            Runnable aFirstDrawCallback = null;
+            if (aSurface != null && !widget.getFirstDraw()) {
+                aFirstDrawCallback = () -> {
+                    widget.setFirstDraw(true);
+                    updateWidget(widget);
+                };
+            }
+
+            widget.setSurface(aSurface, aWidth, aHeight, aFirstDrawCallback);
+
+            View view = (View) widget;
+            // Add widget to a virtual display for invalidation
+            if (aSurface != null && view.getParent() == null) {
+                mWidgetContainer.addView(view, new FrameLayout.LayoutParams(aWidth, aHeight));
+            } else if (aSurface == null && view.getParent() != null) {
+                mWidgetContainer.removeView(view);
+            }
+            view.postInvalidate();
+        });
+    }
+
     @Keep
     @SuppressWarnings("unused")
     void handleMotionEvent(final int aHandle, final int aDevice, final boolean aPressed, final float aX, final float aY) {
         runOnUiThread(() -> {
             Widget widget = mWidgets.get(aHandle);
             if (widget == null) {
                 MotionEventGenerator.dispatch(mRootWidget, aDevice, aPressed, aX, aY);
-
+            } else if (widget == mBrowserWidget && mBrowserWidget.getBorderWidth() > 0) {
+
+                final float extra = mBrowserWidget.getBorderWidth() * 2.0f;
+                final float rx = aX / mBrowserWidget.getTextureWidth();
+                final float ry = aY/ mBrowserWidget.getTextureHeight();
+                final float x = rx * (mBrowserWidget.getTextureWidth() - extra);
+                final float y = ry * (mBrowserWidget.getTextureHeight() - extra);
```

Comment:
Since the pixels are still the same size, I don't think you need to scale them? I think:
```
final int border = mBrowserWidget.getBorderWidth();
MotionEventGenerator.dispatch(widget, aDevice, aPressed, x - border, y - boarder);
```
Since we are just shifting the window by `border` number of pixels. The compositor doesn't know there is a border so pixel size should be unaffected.